iocore/net/NetVConnection.cc

```diff
@@ -118,3 +118,19 @@ NetVCOptions::get_family_string() const
   }
   return {};
 }
+
+void *
+NetVConnection::get_service(const std::type_info &info)
+{
+  if (this->_services.find(std::type_index(info)) != this->_services.end()) {
+    return this;
+  } else {
+    return nullptr;
+  }
+}
```

Comment:
This will not work properly, for reasons illustrated by this example code:  https://godbolt.org/z/bGofcYhse .  Not that f() and g() do not have the same assembly code.  With multiple inheritance, there is no known implementation (and probably impossible) where all the base classes to have the same (start) address as the derived class.  Even with no multiple inheritance, I think it's undefined behavior to assume that (direct and indirect) base classes have the same address as the derived class.  Although the de facto portability of this assumption is very high in that case.

Reply:
Ok, looks like this approach doesn't work at a minimum. What about other approaches? Do you see any technical issues?